frodo.js

```diff
@@ -3,54 +3,57 @@
  *
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.